editor/js/components/arcontroller.component.js

```diff
@@ -71,212 +70,210 @@ ArControllerComponent.prototype.onRemovedFromScene = function( scene ) {
 
 ArControllerComponent.prototype.startAR = function() {
     console.log("Start AR");
-    var self = null;
-
-
-    var texture = undefined;
-    //basic shader
-    //create basic matrices for cameras and transformation
-    var proj = mat4.create();
-    var view = mat4.create();
-    var model = mat4.create();
-    var mvp = mat4.create();
-    var temp = mat4.create();
-
-    //get mouse actions
-                  //set the camera position
-
-
-    //basic phong shader
-    var shader = new Shader('\
-        precision highp float;\
-        attribute vec3 a_vertex;\
-        attribute vec3 a_normal;\
-        attribute vec2 a_coord;\
-        varying vec3 v_normal;\
-        varying vec2 v_coord;\
-        uniform mat4 u_mvp;\
-        uniform mat4 u_model;\
-        void main() {\
-          v_coord = a_coord;\
-          gl_Position =u_mvp* vec4(a_vertex,1.0);\
-        }\
-        ', '\
-        precision highp float;\
-        varying vec3 v_normal;\
-        varying vec2 v_coord;\
-        uniform vec3 u_lightvector;\
-        uniform vec4 u_color;\
-        uniform sampler2D u_texture;\
-        void main() {\
-          vec4 color =  texture2D( u_texture, v_coord);\
-          gl_FragColor = color ;\
-        }\
-      ');
-
-
-    //generic gl flags and settings
+    this._video = this.getUserMedia()
+    this._arTrackable2DList.forEach(arTrackable => {
+      var self = this;
+      window.addEventListener('getDataFromWorker', function(ev) {
+
+        const markerIndex = ev.detail.data.index;
+        const markerType = ev.detail.data.type;
+        const marker = ev.detail.data.marker;
+        trackableId = ev.detail.data.marker.id;
+
+        /*if (ev.detail.data.type === 2) {
+          trackableId = ev.detail.data.marker.id;
+        }*/
+        if (trackableId !== -1) {
+            console.log("saw a trackable with id", trackableId);
+
+            let markerRoot = arTrackable.attachedGameObject;
```

Comment:
as said in the comment probably this is wrong, because referring to a the trackable that not exist. In fact the ArController is now only created in the WebWorker.

Reply:
this is not right. This was cahanged by some changes in litescene.js.